### PR TITLE
feat: persist caption settings in localStorage on track mode change (code100x#1493)

### LINKED ISSUE #1493: feature: Save Caption Settings Between Video Plays
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I have to manually adjust my caption settings every time I play a video. 
It would be much easier if my preferences were saved, as I often prefer captions to be enabled.

**Describe the solution you'd like**
I would like the application to save my caption settings using local storage or cookies so that when I play a video, my preference caption setting is automatically applied without doinng manually.

**Describe alternatives you've considered**
It can be implement using cookies also , but local storage is easy and more starighforward .

**Additional context**

### PR DESCRIPTION
**Pull request by IkramBagban**
…code100x#1493)

### PR Fixes:
- 1 Added functionality to persist subtitle/caption settings to localStorage
- 2 Monitored track mode changes and saved the current mode ('showing' or 'hidden')
- 3 On player initialization, retrieved and applied the saved caption setting from localStorage
- 4 Added cleanup for event listeners to prevent memory leaks

Resolves #1493

### Checklist before requesting a review
- [✓] I have performed a self-review of my code
- [✓] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

File: src/components/VideoPlayer2.tsx [hunk 1]
```diff
@@ -92,6 +92,52 @@ export const VideoPlayer: FunctionComponent<VideoPlayerProps> = ({
     return pipButtonContainer;
   };
 
+  // Fetch previously saved captions setting from localStorage
```
Comment (devsargam) [inline src/components/VideoPlayer2.tsx:95]: Can you remove this comment?

File: src/components/VideoPlayer2.tsx [hunk 2]
```diff
@@ -92,6 +92,52 @@ export const VideoPlayer: FunctionComponent<VideoPlayerProps> = ({
...
+  useEffect(() => {
```
Comment (devsargam) [inline src/components/VideoPlayer2.tsx:96]: Why have 2 useEffects with same dependencies array? Can you add the below login in here only?

**Review by devsargam — CHANGES_REQUESTED** [review]

File: src/components/VideoPlayer2.tsx (hunk 1 above)
Comment (IkramBagban) [reply src/components/VideoPlayer2.tsx:95]: i added this comment to help other people understand these line of codes. anyways i'll remove it

File: src/components/VideoPlayer2.tsx (hunk 2 above)
Comment (IkramBagban) [reply src/components/VideoPlayer2.tsx:96]: yes sure

**Comment by IkramBagban** [comment]
hey @devsargam Please check now

**Comment by devsargam** [comment]
Thank you sir

**Pull request merged**